src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -858,21 +826,20 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
         'by a bug in React. Please file an issue.',
     );
     isPerformingWork = true;
-    const isPerformingDeferredWork = !!deadline;
 
     // This outer loop exists so that we can restart the work loop after
     // catching an error. It also lets us flush Task work at the end of a
     // deferred batch.
-    while (priorityLevel !== NoWork && !fatalError) {
-      invariant(
-        deadline !== null || priorityLevel < HighPriority,
-        'Cannot perform deferred work without a deadline. This error is ' +
-          'likely caused by a bug in React. Please file an issue.',
-      );
-
+    let keepPerformingWork = true;
+    let hasRemainingAsyncWork = false;
+    while (keepPerformingWork && fatalError === null) {
```

Comment:
Basically this is a result of me trying to both 1) avoid `while (true)`, and 2) not repeat the same checks that are done in `workLoop`.

I could get rid of `hasRemainingAsyncWork` by scheduling the callback inside this loop instead of once we exit it. That's how I did it originally but I found this easier to follow. I'll change it back.